src/execution.rs

```diff
@@ -93,6 +96,154 @@ impl File {
 
         Ok(())
     }
+
+    pub fn try_visit_matches<'tree, E, F>(
+        &self,
+        tree: &'tree Tree,
+        source: &'tree str,
+        lazy: bool,
+        mut visit: F,
+    ) -> Result<(), E>
+    where
+        F: FnMut(Match<'_, 'tree>) -> Result<(), E>,
+    {
+        if lazy {
```

Comment:
As an aside, do we need to continue having both strict and lazy evaluation modes?  IIRR, we kept them both as you were first implementing lazy evaluation to give us a gradual migration strategy.  But now that lazy is working in production, do we need to keep strict around?

Reply:
Good question. If I recall, @tausbn was using some mutable state tricks that only work in strict more. Maybe he can weigh in on that?

Reply:
By my count, we have _two_ remaining mutable state tricks left. (See backlink to the relevant issue below.)

I think it should be possible to get rid of them, though that has not been a priority so far.